### PR TITLE
Hw3

### PR DESCRIPTION
**Pull request by lyalchenko1995**

### CODE REVIEW

File: src/test/java/com/epam/tc/hw3/pages/DifferentElementsPage.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+    private LogComponent logComponent;
+
+    @FindBy(xpath = "//label[@class='label-checkbox'][1]/*")
+    private WebElement waterCheckBox;
+
+    @FindBy(xpath = "//label[@class='label-checkbox'][3]/*")
+    private WebElement windCheckBox;
+
+    @FindBy(xpath = "//label[@class='label-radio'][4]/*")
+    private WebElement radioSelen;
```
Comment (komyagin) [inline src/test/java/com/epam/tc/hw3/pages/DifferentElementsPage.java]: It's not a good approach to use indexes in the locators. Only in special situations. And this is not that one

File: src/test/java/com/epam/tc/hw3/pages/components/ColorsDropDownComponent.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+
+public class ColorsDropDownComponent {
+
+    @FindBy(xpath = ".//*[@class='colors']//*[@class='uui-form-element']")
+    private WebElement colorsDropDown;
+
+    @FindBy(xpath = ".//*[@class='colors']//*[@class='uui-form-element']//option[text() = 'Yellow']")
+    private WebElement yellowColor;
```
Comment (komyagin) [inline src/test/java/com/epam/tc/hw3/pages/components/ColorsDropDownComponent.java:12]: There are more suitable solutions and locators

File: src/test/java/com/epam/tc/hw3/pages/components/HeaderMenuComponent.java [hunk 3]
```diff
@@ -0,0 +1,43 @@
+package com.epam.tc.hw3.pages.components;
+
+import java.util.List;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+
+public class HeaderMenuComponent {
+
+    @FindBy(css = "ul.uui-navigation.nav.navbar-nav.m-l8 > li")
```
Comment (komyagin) [inline src/test/java/com/epam/tc/hw3/pages/components/HeaderMenuComponent.java]: Could be simplified to .m-l8 > li

File: src/test/java/com/epam/tc/hw3/pages/components/LogComponent.java [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+public class LogComponent {
+
+    @FindBy(css = "ul[class='panel-body-list logs'] > li")
+    private List<WebElement> logs;
+
+    @FindBy(xpath = "//*[contains(text(),'Water: condition changed to true')]")
+    private WebElement waterLog;
+
+    @FindBy(xpath = "//*[contains(text(),'Wind: condition changed to true')]")
+    private WebElement windLog;
```
Comment (komyagin) [inline src/test/java/com/epam/tc/hw3/pages/components/LogComponent.java]: Could be simplified and made on parametrized method

File: src/test/java/com/epam/tc/hw3/pages/components/LogComponent.java [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+    @FindBy(xpath = "//*[contains(text(),'metal: value changed to  Selen')]")
+    private WebElement metalLog;
+
+    @FindBy(xpath = "//*[contains(text(),'Colors: value changed to Yellow')]")
+    private WebElement colorsLog;
```
Comment (komyagin) [inline src/test/java/com/epam/tc/hw3/pages/components/LogComponent.java]: The same

File: src/test/java/com/epam/tc/hw3/pages/components/SideMenuComponent.java [hunk 6]
```diff
@@ -0,0 +1,15 @@
+package com.epam.tc.hw3.pages.components;
+
+import java.util.List;
+import org.openqa.selenium.WebElement;
+import org.openqa.selenium.support.FindBy;
+
+public class SideMenuComponent {
+
+    @FindBy(css = "ul[class='sidebar-menu left'] > li")
```
Comment (komyagin) [inline src/test/java/com/epam/tc/hw3/pages/components/SideMenuComponent.java]: Could be simplified to .sidebar-menu > li

File: src/test/resources/suiteHW3.xml [hunk 7]
```diff
@@ -0,0 +1,8 @@
+<suite name="suiteHW2" verbose="1" >
+    <test name="Test" >
+        <classes>
+            <class name="com.epam.tc.hw3.scripts.Ex1FromHW2"/>
+            <class name="com.epam.tc.hw3.scripts.Ex2FromHW2"/>
+        </classes>
+    </test>
+</suite>
```
Comment (komyagin) [inline src/test/resources/suiteHW3.xml]: Add a new blank line to the end of the file
image

**Review by komyagin — CHANGES_REQUESTED** [review]

File: src/test/java/com/epam/tc/hw3/pages/DifferentElementsPage.java (hunk 1 above)
Comment (lyalchenko1995) [reply src/test/java/com/epam/tc/hw3/pages/DifferentElementsPage.java]: done

File: src/test/java/com/epam/tc/hw3/pages/components/HeaderMenuComponent.java (hunk 3 above)
Comment (lyalchenko1995) [reply src/test/java/com/epam/tc/hw3/pages/components/HeaderMenuComponent.java]: done

File: src/test/java/com/epam/tc/hw3/pages/components/SideMenuComponent.java (hunk 6 above)
Comment (lyalchenko1995) [reply src/test/java/com/epam/tc/hw3/pages/components/SideMenuComponent.java]: done

File: src/test/resources/suiteHW3.xml (hunk 7 above)
Comment (lyalchenko1995) [reply src/test/resources/suiteHW3.xml]: done

File: src/test/java/com/epam/tc/hw3/pages/components/ColorsDropDownComponent.java (hunk 2 above)
Comment (lyalchenko1995) [reply src/test/java/com/epam/tc/hw3/pages/components/ColorsDropDownComponent.java:12]: done

File: src/test/java/com/epam/tc/hw3/pages/components/LogComponent.java (hunk 4 above)
Comment (lyalchenko1995) [reply src/test/java/com/epam/tc/hw3/pages/components/LogComponent.java]: done

File: src/test/java/com/epam/tc/hw3/pages/components/LogComponent.java (hunk 5 above)
Comment (lyalchenko1995) [reply src/test/java/com/epam/tc/hw3/pages/components/LogComponent.java]: done

**Review by komyagin — COMMENTED** [review]
Changed according to comments

**Pull request merged**